app/CMakeLists.txt

```diff
@@ -366,12 +352,26 @@ if(CELERITAS_BUILD_DEMOS AND CELERITAS_USE_Geant4)
   set(_demo_geant_integration_libs Celeritas::accel)
 
   if(CELERITAS_USE_ROOT)
-    celeritas_reflex_generate_dictionary(HitClasses
-      demo-geant-integration/HitClasses.hh
-      SELECTION demo-geant-integration/selection.xml
+    include_directories(
+      # root_generate_dictionary can not handle the '$<INSTALL_INTERFACE:include>'
+      # part of the Celeritas::corecel target's include, so we can not use it in
+      # the 'DEPENDENCIES' property.
+      "${CELERITAS_HEADER_CONFIG_DIRECTORY}"
+      "${PROJECT_SOURCE_DIR}/src"
+      # The include directory are attached to the Geant4 target through INTERFACE_INCLUDE_DIRECTORIES
+      # which is not looked at by root_generate_dictionary.
+      ${Geant4_INCLUDE_DIRS}
```

Comment:
`${CLHEP_INCLUDE_DIRS}` needs to be reinstated because `${Geant4_INCLUDE_DIRS}` doesn't propagate it. There's an ugly hack in `GeantConfig.cmake` that globally includes CLHEP "in case ROOT needs it" but to avoid unnecessary includes (and catch potential issues with missing includes in our targets) I explicitly undo that in our top-level CMakeLists.

Reply:
Maybe there's something about your build that has the geant4 and clhep headers living side by side. It fails on the CI because of this.